### PR TITLE
[E2E] Introduce `visitDashboard` helper

### PR DESCRIPTION
**Pull request by nemanjaglumac**
### Status
READY

### What does this PR accomplish?
- Adds new `visitDashboard` E2E helper whose aim is to simplify code and to greatly reduce the overall flakiness
- It automatically waits for the last request* when one visits the dashboard
    - *it depends whether that user has the permissions to view the dashboard or not, but helper takes that into account
- Applies new helper to all related tests

### CODE REVIEW

File: frontend/test/__support__/e2e/helpers/e2e-misc-helpers.js [hunk 1]
```diff
@@ -150,3 +150,30 @@ export function visitQuestion(id) {
...
+  cy.intercept("GET", `/api/dashboard/${id}`).as("getDashboard");
+  // The very last request when visiting dashboard always checks the collection it is in.
+  // That is - IF user has the permission to view that dashboard!
+  cy.intercept("GET", `/api/collection/*`).as("getParentCollection");
+
+  cy.visit(`/dashboard/${id}`);
+
+  // If users doesn't have permissions to even view the dashboard,
+  // the last request for them would be `getDashboard`.
+  cy.wait("@getDashboard").then(({ response: { statusCode } }) => {
```
Comment (nemanjaglumac) [inline frontend/test/__support__/e2e/helpers/e2e-misc-helpers.js:172]: For example:
image

Prior to this adjustment, [this test](https://github.com/metabase/metabase/blob/f3432569e8665044577c62e2ddf95f371031987b/frontend/test/metabase/scenarios/dashboard/permissions.cy.spec.js#L115) failed
https://app.circleci.com/pipelines/github/metabase/metabase/31571/workflows/be70cbef-a36e-49db-8477-f69c4ccbb2fd/jobs/1601335/artifacts

File: frontend/test/metabase/scenarios/dashboard/reproductions/20637-add-series-to-dashcard.cy.spec.js [hunk 2]
```diff
@@ -86,7 +86,7 @@ function createQuestionsAndDashboard() {
             ],
           });
 
-          visitDashboard(dashboard_id);
+          cy.visit(`/dashboard/${dashboard_id}`);
```
Comment (nemanjaglumac) [inline frontend/test/metabase/scenarios/dashboard/reproductions/20637-add-series-to-dashcard.cy.spec.js:89]: We can always expand the `visitDashboard` helper to account for ALL individual `dashcard` requests, but the code gets complicated really quickly. Without this adjustment, about 99.9% of all tests work just fine.

For cases where we explicitly need to assert on the individual requests, simply do not use the helper.

**Review by ranquild — APPROVED** [review]

**Review by gusaiani — APPROVED** [review]
Yeah, when you mentioned `visitQuestion` I did look for `visitDashboard`, `visitCollection` etc.

**Comment by nemanjaglumac** [comment]
> Yeah, when you mentioned `visitQuestion` I did look for `visitDashboard`, `visitCollection` etc.

`visitCollection` is the next in line :)

**Pull request merged**